clippy_lints/src/methods/filter_next.rs

```diff
@@ -10,51 +10,68 @@ use rustc_span::sym;
 
 use super::FILTER_NEXT;
 
-/// lint use of `filter().next()` for `Iterators`
+#[derive(Copy, Clone)]
+pub(super) enum Direction {
+    Forward,
+    Backward,
+}
+
+/// lint use of `filter().next()` for `Iterator` and `filter().next_back()` for
+/// `DoubleEndedIterator`
 pub(super) fn check<'tcx>(
     cx: &LateContext<'tcx>,
     expr: &'tcx hir::Expr<'_>,
     recv: &'tcx hir::Expr<'_>,
     filter_arg: &'tcx hir::Expr<'_>,
+    direction: Direction,
 ) {
-    // lint if caller of `.filter().next()` is an Iterator
-    let recv_impls_iterator = cx
+    // lint if caller of `.filter().next()` is an Iterator or `.filter().next_back()` is a
+    // DoubleEndedIterator
+    let (required_trait, (next_method, find_method)) = match direction {
+        Direction::Forward => (sym::Iterator, ("next", "find")),
+        Direction::Backward => (sym::DoubleEndedIterator, ("next_back", "rfind")),
+    };
```

Comment:
nit: The nested tuple isn't adding any clarity. Just use `(required_trait, next_method, find_method)`.